### PR TITLE
lib: resolve NYIs from #3524 by renameing features as specified

### PR DESCRIPTION
**Pull request by simonvonhackewitz**
Sequence.fz
* `is_prefix_of` -> `starts_with`
* `find_first` -> `find`
* `replace_all` -> `replace`
* `replace_n` -> `replace`
* `split_` -> `split0`

### CODE REVIEW

File: lib/Sequence.fz [hunk 1]
```diff
@@ -613,18 +613,16 @@ public Sequence(public T type) ref is
...
   =>
-    match as_list
+    match l.as_list
       nil     => true
       c1 Cons =>
-        match l.as_list
+        match as_list
           nil     => false
-          c2 Cons => c1.head = c2.head && c1.tail.is_prefix_of c2.tail # tail recursion
+          c2 Cons => c2.head = c1.head && c2.tail.starts_with c1.tail # tail recursion
```
[the remark this answers is not in the document]
Comment (michaellilltokiwa) [reply lib/Sequence.fz:625]: @maxteufel Are you referring to replacing `is_prefix_of` by `starts_with` or where do you see reversal of logic?

[the remark this answers is not in the document]
Comment (michaellilltokiwa) [reply lib/Sequence.fz:625]: @maxteufel Nevermind, c1 and c2 have been replaced. I see it now as well...

File: lib/Sequence.fz [hunk 2]
```diff
@@ -721,7 +713,7 @@ public Sequence(public T type) ref is
   =>
     match as_list
       nil     => 0
-      c1 Cons => (if (l.is_prefix_of Sequence.this) 1 + (drop l.count).count_matches l
+      c1 Cons => (if (starts_with l) 1 + (drop l.count).count_matches l
                   else                                  c1.tail       .count_matches l)
```
Comment (michaellilltokiwa) [inline lib/Sequence.fz:717]: Does this work?
```suggestion
      c1 Cons => if starts_with l then 1 + (drop l.count).count_matches l
                   else                                  c1.tail       .count_matches l
```

Comment (simonvonhackewitz) [reply lib/Sequence.fz]: yes, it seems to work

**Pull request merged**